daemon/daemon.go

```diff
@@ -905,6 +883,51 @@ func (d *Daemon) init() error {
 	return nil
 }
 
+// startLXCMapSync spawns a controller which synchronizes the host entries in BPF lxc Map
+// with the in-memory host IP address values.
+func (d *Daemon) startLXCMapSync() error {
+	// Initially, clean all endpoint entries
+	if err := lxcmap.LXCMap.DeleteAll(); err != nil {
+		return err
+	}
+
+	controller.NewManager().UpdateController("lxcmap-bpf-host-sync",
+		controller.ControllerParams{
+			DoFunc: func() error {
+				// TODO: Update addresses first, in case node addressing has changed.
+				// TODO: Once these start changing on runtime, figure out the locking strategy.
+				// TODO: Delete stale host entries from the lxcmap.
+				localIPs := []net.IP{
+					node.GetInternalIPv4(),
+					node.GetExternalIPv4(),
+					node.GetIPv6(),
+					node.GetIPv6Router(),
+				}
+
+				for _, ip := range localIPs {
+					if err := lxcmap.SyncHostEntry(ip); err != nil {
+						return fmt.Errorf("Unable to add host entry to endpoint map: %s", err)
+					}
+					id, exists := ipcache.IPIdentityCache.LookupByIP(ip.String())
+					if !exists || id != identity.ReservedIdentityHost {
```

Comment:
If the ipcache thinks that the identity for the IP is not `reserved:host`, then this will trigger. Then presumably re-trigger, again and again as long as the entry exists in the ipcache?